code/game/turfs/unsimulated/snow.dm

```diff
@@ -0,0 +1,226 @@
+			adj_tile.update_icon()
+
+/turf/unsimulated/floor/snow/attackby(obj/item/weapon/W as obj, mob/user as mob)
+
+	..()
+
+	if(istype(W, /obj/item/weapon/pickaxe/shovel))
+		user.visible_message("<span class='notice'>[user] starts digging out some snow with \the [W].</span>", \
+		"<span class='notice'>You start digging out some snow with \the [W].</span>")
+		user.delayNextAttack(20)
+		if(do_after(user, src, 20))
+			user.visible_message("<span class='notice'>[user] digs out some snow with \the [W].</span>", \
+			"<span class='notice'>You dig out some snow with \the [W].</span>")
+			extract_snowballs(5, 0, user)
+
+/turf/unsimulated/floor/snow/attack_hand(mob/user as mob)
+
+	//Reach down and make a snowball
+	user.visible_message("<span class='notice'>[user] reaches down and starts forming a snowball.</span>", \
+	"<span class='notice'>You reach down and start forming a snowball.</span>")
+	user.delayNextAttack(10)
+	if(do_after(user, src, 5))
+		user.visible_message("<span class='notice'>[user] finishes forming a snowball.</span>", \
+		"<span class='notice'>You finish forming a snowball.</span>")
+		extract_snowballs(1, 1, user)
+
+	..()
+
+/turf/unsimulated/floor/snow/proc/extract_snowballs(var/snowball_amount = 0, var/pick_up = 0, var/mob/user)
+
+	if(!snowball_amount)
+		return
+
+	var/extract_amount = min(snowballs, snowball_amount)
+
+	for(var/i = 0; i < extract_amount, i++)
+		var/obj/item/stack/sheet/snow/snowball = new /obj/item/stack/sheet/snow(user.loc)
+		snowball.pixel_x = rand(-16, 16) //Would be wise to move this into snowball New() down the line
+		snowball.pixel_y = rand(-16, 16)
+
+		if(pick_up)
+			user.put_in_hands(snowball)
+
+		snowballs--
+
+	if(!snowballs) //We're out of snow, turn into a permafrost tile
+		ChangeTurf(/turf/unsimulated/floor/snow/permafrost)
+
+//In the future, catwalks should be the base to build in the arctic, not lattices
+//This would however require a decent rework of floor construction and deconstruction
+/turf/unsimulated/floor/snow/permafrost/canBuildCatwalk()
+	return BUILD_FAILURE
+
+/turf/unsimulated/floor/snow/permafrost/canBuildLattice()
+	if(x >= (world.maxx - TRANSITIONEDGE) || x <= TRANSITIONEDGE)
+		return BUILD_FAILURE
+	else if (y >= (world.maxy - TRANSITIONEDGE || y <= TRANSITIONEDGE ))
+		return BUILD_FAILURE
+	else if(!(locate(/obj/structure/lattice) in contents))
+		return BUILD_SUCCESS
+	return BUILD_FAILURE
+
+/turf/unsimulated/floor/snow/permafrost/canBuildPlating()
+	if(x >= (world.maxx - TRANSITIONEDGE) || x <= TRANSITIONEDGE)
+		return BUILD_FAILURE
+	else if (y >= (world.maxy - TRANSITIONEDGE || y <= TRANSITIONEDGE ))
+		return BUILD_FAILURE
+	else if(locate(/obj/structure/lattice) in contents)
+		return BUILD_SUCCESS
+	return BUILD_FAILURE
+
+/turf/unsimulated/floor/snow/Entered(mob/user)
+	..()
+	if(isliving(user) && !user.locked_to && !user.lying && !user.flying)
+		playsound(get_turf(src), pick(snowsound), 10, 1, -1, channel = 123)
```

Comment:
It's to prevent the sounds of moving in snow from stacking over each other creating an awful cacophony